tfrecord/reader.py

```diff
@@ -90,9 +90,11 @@ def read_records(start_offset=None, end_offset=None):
 
 
 def tfrecord_loader(data_path: str,
-                    index_path: str,
+                    index_path: typing.Union[str, None],
                     description: typing.Union[typing.List[str], typing.Dict[str, str], None] = None,
-                    shard: typing.Optional[typing.Tuple[int, int]] = None
+                    shard: typing.Optional[typing.Tuple[int, int]] = None,
+                    transform_func: typing.Callable[[dict], typing.Any] = None,
```

Comment:
Don't add this to tfrecord_loader. It should only be available in PyTorch datasets.

Reply:
I'm not sure if I understand. How we can only add in PyTorch dataset and not to tfrecord_loader?

In PyTorch dataset we yield an iterator which we get from `reader.tfrecord_loader`, so we need to make changes there, right?

Reply:
https://docs.python.org/3/library/functions.html#map

Reply:
Aah, that's a good use of map. Thanks! Updated code.

JW, what's the reason of not having it in `tfrecord_loader` too? I don't use `tfrecord_loader` almost ever, except when I am debugging.

Reply:
Because tfrecord_loader is a low level API.